modules/tinymce/src/plugins/table/main/ts/selection/SelectionTargets.ts

```diff
@@ -53,6 +54,46 @@ interface ExtractedSelectionDetails {
 
 type TargetSetupCallback = (targets: RunOperation.CombinedTargets) => boolean;
 
+const isElementHeader = (cell: SugarElement<HTMLTableCellElement>) => {
+  if (SugarNode.name(cell) === 'th') {
+    return true;
+  } else {
+    return SelectorFind.ancestor(cell, 'thead').isSome();
+  }
+};
+
+export const isEntireRowsHeaders = (editor: Editor, selections: Selections) => {
+  const startCellOpt = TableSelection.getSelectionStartCell(Util.getSelectionStart(editor));
+
+  return startCellOpt.exists((startCell) => {
+    const tableOpt = TableLookup.table(startCell, Util.getIsRoot(editor));
+
+    return tableOpt.exists((table) => {
+      const warehouse = Warehouse.fromTable(table);
+      const targets = TableTargets.forMenu(selections, table, startCell);
+
+      const usedRows: number[] = [];
+      Arr.each(warehouse.all, (row) => {
+        Arr.each(row.cells, (cell) => {
+          const existsInSelection = Arr.exists(targets.selection, (element) =>
+            element.dom === cell.element.dom
+          );
+
+          if (existsInSelection && !Arr.contains(usedRows, cell.row)) {
+            usedRows.push(cell.row);
+          }
+        });
+      });
+
+      return Arr.forall(usedRows, (rowIndex) =>
+        Arr.forall(warehouse.all[rowIndex].cells, (cell) =>
+          isElementHeader(cell.element)
+        )
+      );
+    });
+  });
+};
```

Comment:
Does this function need to be this complex? I might be misunderstanding but I believe it's checking if all cells within the selection exist within a header row or not? Do we need to check every cell in that row or can we safely assume that the state of the table is correct and than if one cell in a row is in a `thead` then they all are?

```suggestion
export const isEntireRowsHeaders = (editor: Editor, selections: Selections) => {
  const selectedCells = TableSelection.getCellsFromSelection(Util.getSelectionStart(editor), selections, Util.getIsRoot(editor));
  return Arr.forall(selectedCells, isElementHeader);
};
```

Reply:
As mentioned in my overall comment, we shouldn't even have a function for this as the logic already exists, so this is duplicate code.